src/main/java/ua/artcode/billapp/security/JwtValidator.java

```diff
@@ -0,0 +1,34 @@
+package ua.artcode.billapp.security;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import ua.artcode.billapp.model.JwtUser;
+
+@Component
+public class JwtValidator {
+
+    @Value("${billapp.token.secret}")
+    private String secret;
+
+
+    public JwtUser validate(String token) {
+
+        JwtUser jwtUser = null;
+
+        try {
+            Claims body = Jwts.parser()
+                    .setSigningKey(secret)
+                    .parseClaimsJws(token)
+                    .getBody();
+            jwtUser = new JwtUser();
+            jwtUser.setPhone(body.getSubject());
+            jwtUser.setId(Long.parseLong((String) body.get("userId")));
+            jwtUser.setRole((String) body.get("role"));
+        } catch (Exception e){
```

Comment:
If validation is failed, it is better to throw (without catch) some AuthorizationException (NOT RUNTIME), or return proper response with 403 status code.

Reply:
Thank you, I will fix it.